app/controllers/registrations_controller.rb

```diff
@@ -3,7 +3,7 @@ class RegistrationsController < Devise::RegistrationsController
   private
 
   def sign_up_params
-    params.require(:user).permit(:name, :email, :password, :password_confirmation)
+    params.require(:user).permit(:name, :email, :password, :password_confirmation).merge(:locale => I18n.locale)
```

Comment:
as part of this PR, change `before_action :set_locale` with `around_action :set_locale`, and the following implementation:

```ruby
  def set_locale
    I18n.locale = params[:locale] || I18n.default_locale
    yield
    I18n.locale = I18n.default_locale
  end
```

The point is to not let Arabic users change the global locale for all users.